### PR TITLE
Webflow V2 migration

### LINKED ISSUE #14688: [Components] Webflow v2
Webflow is sunsetting their v1 APIs as of January 1, 2025. 

We need to migrate the majority of existing components over to their v2 APIs, which include a few changes. I've created a separate app here, `webflow_v2` as it requires a new client with updated access tokens to access the v2 APIs. I've also filtered the actions down to those with the most usage as a starting point.

There are some breaking changes between the V1 API and V2 API.
https://www.npmjs.com/package/webflow-api

**Triggers**:
https://developers.webflow.com/v1.0.0/reference/sites/webhooks/create-webhook

Please create webhook triggers for:
- New Form Submission - use async options to optionally filter form
- New Site Published 
- New Order Placed
- Updated Order 
- New Change in Inventory
- New Collection Item Created
- Collection Item Updated
- Collection Item Deleted

**Actions**:
- Create Collection Item
- Update Collection Item
- List Collection Items
- Get Collection Item
- Publish Site
- Get Collection
- List Collections
- Update Item Inventory
- Fulfill Order
- Update Order
- Delete Collection Item

### PR DESCRIPTION
**Pull request by GTFalcao**
Closes #14688 

## Release Notes

- **New Features**
  - Introduced new actions for managing orders, sites, collections, and items within the Webflow integration.
  - Added methods for retrieving, creating, updating, and deleting collection items and orders.
  - New properties for domains, sites, collections, items, and orders have been added to enhance functionality.
  - New actions for fulfilling, unfulfilling, and refunding orders added to streamline order management.
  - New action for retrieving order details introduced.
  - New action for publishing sites added.
  - Added a new action for updating collection items.
  - New module for handling e-commerce order events and form submissions.
  - New module for handling new collection item events and deleted collection items.
  - New sources for new e-commerce orders, form submissions, and site published events.

- **Improvements**
  - Transitioned to a new application structure, simplifying API interactions and improving code clarity.
  - Versioning updated to a dynamic format for better tracking.
  - Enhanced webhook management with new methods for creating and removing webhooks.
  - Streamlined the interaction with the Webflow API by consolidating API calls under a unified application interface.

- **Bug Fixes**
  - Removed obsolete actions to streamline the integration and improve performance.
  - Improved error handling and response management for API calls.

### CODE REVIEW

File: components/webflow_v2/actions/create-collection-item/create-collection-item.mjs [hunk 1]
```diff
@@ -1,44 +1,38 @@
-import webflow from "../../webflow.app.mjs";
+import app from "../../webflow_v2.app.mjs";
 
 export default {
-  key: "webflow-create-collection-item",
+  key: "webflow_v2-create-collection-item",
   name: "Create Collection Item",
   description: "Create new collection item. [See the docs here](https://developers.webflow.com/#create-new-collection-item)",
-  version: "0.1.7",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/actions/create-collection-item/create-collection-item.mjs:7]: Please set the right version

File: components/webflow_v2/actions/delete-collection-item/delete-collection-item.mjs [hunk 2]
```diff
@@ -1,22 +1,22 @@
-import webflow from "../../webflow.app.mjs";
+import app from "../../webflow_v2.app.mjs";
 
 export default {
-  key: "webflow-delete-collection-item",
+  key: "webflow_v2-delete-collection-item",
   name: "Delete Collection Item",
   description: "Delete Item of a Collection. [See the docs here](https://developers.webflow.com/#remove-collection-item)",
-  version: "0.0.5",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/actions/delete-collection-item/delete-collection-item.mjs:7]: Please set the right version

File: components/webflow_v2/actions/get-collection-item/get-collection-item.mjs [hunk 3]
```diff
@@ -1,22 +1,22 @@
-import webflow from "../../webflow.app.mjs";
+import app from "../../webflow_v2.app.mjs";
 
 export default {
-  key: "webflow-get-collection-item",
+  key: "webflow_v2-get-collection-item",
   name: "Get Collection Item",
   description: "Get a Collection Item. [See the docs here](https://developers.webflow.com/#get-single-item)",
-  version: "0.1.7",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/actions/get-collection-item/get-collection-item.mjs:7]: Please set the right version

File: components/webflow_v2/actions/get-collection/get-collection.mjs [hunk 4]
```diff
@@ -1,22 +1,22 @@
-import webflow from "../../webflow.app.mjs";
+import app from "../../webflow_v2.app.mjs";
 
 export default {
-  key: "webflow-get-collection",
+  key: "webflow_v2-get-collection",
   name: "Get Collection",
   description: "Get a collection. [See the docs here](https://developers.webflow.com/#get-collection-with-full-schema)",
-  version: "0.0.5",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/actions/get-collection/get-collection.mjs:7]: Please set the right version

File: components/webflow_v2/actions/get-item-inventory/get-item-inventory.mjs [hunk 5]
```diff
@@ -1,22 +1,22 @@
-import webflow from "../../webflow.app.mjs";
+import app from "../../webflow_v2.app.mjs";
 
 export default {
-  key: "webflow-get-item-inventory",
+  key: "webflow_v2-get-item-inventory",
   name: "Get Item Inventory",
   description: "Get the inventory of a specific item. [See the docs here](https://developers.webflow.com/#item-inventory)",
-  version: "0.0.5",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/actions/get-item-inventory/get-item-inventory.mjs:7]: Please set the right version

File: components/webflow_v2/actions/get-order/get-order.mjs [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import app from "../../webflow_v2.app.mjs";
+
+export default {
+  key: "webflow_v2-get-order",
+  name: "Get Order",
+  description: "Get info on an order. [See the docs here](https://developers.webflow.com/#get-order)",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/actions/get-order/get-order.mjs:7]: Please set the right version

File: components/webflow_v2/actions/get-site/get-site.mjs [hunk 7]
```diff
@@ -1,22 +1,22 @@
-import webflow from "../../webflow.app.mjs";
+import app from "../../webflow_v2.app.mjs";
 
 export default {
-  key: "webflow-get-site",
+  key: "webflow_v2-get-site",
   name: "Get Site",
   description: "Get a site. [See the docs here](https://developers.webflow.com/#get-specific-site)",
-  version: "0.0.4",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/actions/get-site/get-site.mjs:7]: Please set the right version

File: components/webflow_v2/actions/list-collection-items/list-collection-items.mjs [hunk 8]
```diff
@@ -1,22 +1,22 @@
-import webflow from "../../webflow.app.mjs";
+import app from "../../webflow_v2.app.mjs";
 
 export default {
-  key: "webflow-list-collection-items",
+  key: "webflow_v2-list-collection-items",
   name: "List Collection Items",
   description: "List Items of a collection. [See the docs here](https://developers.webflow.com/#get-all-items-for-a-collection)",
-  version: "0.0.5",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/actions/list-collection-items/list-collection-items.mjs:7]: Please set the right version

File: components/webflow_v2/actions/list-collections/list-collections.mjs [hunk 9]
```diff
@@ -1,22 +1,22 @@
-import webflow from "../../webflow.app.mjs";
+import app from "../../webflow_v2.app.mjs";
 
 export default {
-  key: "webflow-list-collections",
+  key: "webflow_v2-list-collections",
   name: "List Collections",
   description: "List collections. [See the docs here](https://developers.webflow.com/#list-collections)",
-  version: "0.0.4",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/actions/list-collections/list-collections.mjs:7]: Please set the right version

File: components/webflow_v2/actions/list-orders/list-orders.mjs [hunk 10]
```diff
@@ -1,35 +1,35 @@
...
+import app from "../../webflow_v2.app.mjs";
 import constants from "../common/constants.mjs";
 
 export default {
-  key: "webflow-list-orders",
+  key: "webflow_v2-list-orders",
   name: "List Orders",
   description: "List orders. [See the docs here](https://developers.webflow.com/#get-all-orders)",
-  version: "0.0.4",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/actions/list-orders/list-orders.mjs:8]: Please set the right version

File: components/webflow_v2/actions/list-sites/list-sites.mjs [hunk 11]
```diff
@@ -1,16 +1,16 @@
-import webflow from "../../webflow.app.mjs";
+import app from "../../webflow_v2.app.mjs";
 
 export default {
-  key: "webflow-list-sites",
+  key: "webflow_v2-list-sites",
   name: "List Sites",
   description: "List sites. [See the docs here](https://developers.webflow.com/#list-sites)",
-  version: "0.0.4",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/actions/list-sites/list-sites.mjs:7]: Please set the right version

File: components/webflow_v2/actions/publish-site/publish-site.mjs [hunk 12]
```diff
@@ -0,0 +1,34 @@
+import app from "../../webflow_v2.app.mjs";
+
+export default {
+  key: "webflow_v2-publish-site",
+  name: "Publish Site",
+  description: "Publish a site. [See the docs here](https://developers.webflow.com/#publish-site)",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/actions/publish-site/publish-site.mjs:7]: Please set the right version

File: components/webflow_v2/actions/refund-order/refund-order.mjs [hunk 13]
```diff
@@ -1,30 +1,28 @@
-import webflow from "../../webflow.app.mjs";
+import app from "../../webflow_v2.app.mjs";
 
 export default {
-  key: "webflow-refund-order",
+  key: "webflow_v2-refund-order",
   name: "Refund Order",
   description: "Refund an order. [See the docs here](https://developers.webflow.com/#refund-order)",
-  version: "0.0.4",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/actions/refund-order/refund-order.mjs:7]: Please set the right version

File: components/webflow_v2/actions/unfulfill-order/unfulfill-order.mjs [hunk 14]
```diff
@@ -1,30 +1,28 @@
-import webflow from "../../webflow.app.mjs";
+import app from "../../webflow_v2.app.mjs";
 
 export default {
-  key: "webflow-unfulfill-order",
+  key: "webflow_v2-unfulfill-order",
   name: "Unfulfill Order",
   description: "Unfulfill an order. [See the docs here](https://developers.webflow.com/#unfulfill-order)",
-  version: "0.0.4",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/actions/unfulfill-order/unfulfill-order.mjs:7]: Please set the right version

File: components/webflow_v2/actions/update-collection-item/update-collection-item.mjs [hunk 15]
```diff
@@ -0,0 +1,94 @@
+import app from "../../webflow_v2.app.mjs";
+
+export default {
+  key: "webflow_v2-update-collection-item",
+  name: "Update Collection Item",
+  description:
+    "Update collection item. [See the documentation](https://developers.webflow.com/#update-collection-item)",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/actions/update-collection-item/update-collection-item.mjs:8]: Please set the right version

File: components/webflow_v2/actions/update-item-inventory/update-item-inventory.mjs [hunk 16]
```diff
@@ -1,22 +1,22 @@
-import webflow from "../../webflow.app.mjs";
+import app from "../../webflow_v2.app.mjs";
 
 export default {
-  key: "webflow-update-item-inventory",
+  key: "webflow_v2-update-item-inventory",
   name: "Update Item Inventory",
   description: "Update the inventory of a specific item. [See the docs here](https://developers.webflow.com/#update-item-inventory)",
   version: "0.0.5",
```
Comment (luancazarine) [inline components/webflow_v2/actions/update-item-inventory/update-item-inventory.mjs:7]:
```suggestion
  version: "0.0.6",
```

File: components/webflow_v2/actions/update-order/update-order.mjs [hunk 17]
```diff
@@ -1,22 +1,22 @@
-import webflow from "../../webflow.app.mjs";
+import app from "../../webflow_v2.app.mjs";
 
 export default {
-  key: "webflow-update-order",
+  key: "webflow_v2-update-order",
   name: "Update Order",
   description: "Update an order. [See the docs here](https://developers.webflow.com/#update-order)",
-  version: "0.0.4",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/actions/update-order/update-order.mjs:7]: Please set the right version

File: components/webflow_v2/sources/changed-collection-item/changed-collection-item.mjs [hunk 18]
```diff
@@ -2,22 +2,25 @@ import common from "../common/collection-common.mjs";
 
 export default {
   type: "source",
-  key: "webflow-changed-collection-item",
-  name: "New Changed Collection Item",
+  key: "webflow_v2-changed-collection-item",
+  name: "Collection Item Updated",
   description: "Emit new event when a collection item is changed. [See the docs here](https://developers.webflow.com/#model16)",
-  version: "0.2.3",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/sources/changed-collection-item/changed-collection-item.mjs:8]: Please set the right version

File: components/webflow_v2/sources/changed-ecommerce-inventory/changed-ecommerce-inventory.mjs [hunk 19]
```diff
@@ -2,23 +2,24 @@ import common from "../common/common.mjs";
 
 export default {
   type: "source",
-  key: "webflow-changed-ecommerce-inventory",
-  name: "New Changed E-commerce Inventory",
+  key: "webflow_v2-changed-ecommerce-inventory",
+  name: "E-commerce Inventory Updated",
   description: "Emit new event when an e-commerce inventory level changes. [See the docs here](https://developers.webflow.com/#item-inventory)",
-  version: "0.2.3",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/sources/changed-ecommerce-inventory/changed-ecommerce-inventory.mjs:8]: Please set the right version

File: components/webflow_v2/sources/new-site-published/new-site-published.mjs [hunk 20]
```diff
@@ -3,20 +3,17 @@ import common from "../common/common.mjs";
 
 export default {
   type: "source",
-  key: "webflow-new-site-published",
+  key: "webflow_v2-new-site-published",
   name: "New Site Published",
   description: "Emit new event when a site is published. [See the docs here](https://developers.webflow.com/#trigger-types)",
-  version: "0.2.3",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/sources/new-site-published/new-site-published.mjs:9]: Please set the right version

File: components/webflow_v2/sources/new-form-submission/new-form-submission.mjs [hunk 21]
```diff
@@ -0,0 +1,28 @@
+import common from "../common/common.mjs";
+import sampleEmit from "./test-event.mjs";
+
+export default {
+  type: "source",
+  key: "webflow_v2-new-form-submission",
+  name: "New Form Submission",
+  description: "Emit new event when a form is submitted. [See the docs here](https://developers.webflow.com/#trigger-types)",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/sources/new-form-submission/new-form-submission.mjs:9]: Please set the right version

File: components/webflow_v2/sources/new-ecommerce-order/new-ecommerce-order.mjs [hunk 22]
```diff
@@ -2,10 +2,10 @@ import common from "../common/common.mjs";
 
 export default {
   type: "source",
-  key: "webflow-new-ecommerce-order",
+  key: "webflow_v2-new-ecommerce-order",
   name: "New E-commerce Order",
   description: "Emit new event when an e-commerce order is created. [See the docs here](https://developers.webflow.com/#order-model)",
-  version: "0.2.3",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/sources/new-ecommerce-order/new-ecommerce-order.mjs:8]: Please set the right version

File: components/webflow_v2/sources/new-deleted-collection-item/new-deleted-collection-item.mjs [hunk 23]
```diff
@@ -2,21 +2,22 @@ import common from "../common/common.mjs";
 
 export default {
   type: "source",
-  key: "webflow-new-deleted-collection-item",
-  name: "New Deleted Collection Item",
+  key: "webflow_v2-new-deleted-collection-item",
+  name: "Collection Item Deleted",
   description: "Emit new event when a collection item is deleted. [See the docs here](https://developers.webflow.com/#item-model)",
-  version: "0.2.3",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/sources/new-deleted-collection-item/new-deleted-collection-item.mjs:8]: Please set the right version

File: components/webflow_v2/sources/new-collection-item/new-collection-item.mjs [hunk 24]
```diff
@@ -0,0 +1,26 @@
+import common from "../common/common.mjs";
+
+export default {
+  type: "source",
+  key: "webflow_v2-new-collection-item",
+  name: "New Collection Item Created",
+  description: "Emit new event when a collection item is created. [See the docs here](https://developers.webflow.com/#item-model)",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/sources/new-collection-item/new-collection-item.mjs:8]: Please set the right version

File: components/webflow_v2/sources/changed-ecommerce-order/changed-ecommerce-order.mjs [hunk 25]
```diff
@@ -2,23 +2,22 @@ import common from "../common/common.mjs";
 
 export default {
   type: "source",
-  key: "webflow-changed-ecommerce-order",
-  name: "New Changed E-commerce Order",
+  key: "webflow_v2-changed-ecommerce-order",
+  name: "E-commerce Order Updated",
   description: "Emit new event when an e-commerce order is changed. [See the docs here](https://developers.webflow.com/#order-model)",
-  version: "0.2.3",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/webflow_v2/sources/changed-ecommerce-order/changed-ecommerce-order.mjs:8]: Please set the right version

**Review by luancazarine — CHANGES_REQUESTED** [review]
Hi @GTFalcao, I just added some suggestions about versions.

**Comment by GTFalcao** [comment]
Thanks for the notes @luancazarine , the version numbers were left as-is temporarily to speed up testing on this, and I'll set them all to 0.0.1 when the migration is ready to be released

**Review by luancazarine — APPROVED** [review]

**Pull request merged**